astropy/utils/iers/iers.py

```diff
@@ -669,30 +669,20 @@ def open(cls):
             if cls.iers_table.meta.get('data_url') in all_urls:
                 return cls.iers_table
 
-        dl_success = False
-        err_list = []
-
-        for url in all_urls:
-            try:
-                filename = download_file(url, cache=True)
-            except Exception as err:
-                err_list.append(str(err))
-            else:
-                dl_success = True
-                break
-
-        if not dl_success:
+        try:
+            filename = download_file(all_urls[0], sources=all_urls, cache=True)
+        except Exception as err:
             # Issue a warning here, perhaps user is offline.  An exception
             # will be raised downstream when actually trying to interpolate
             # predictive values.
-            warn(AstropyWarning('failed to download {}, using local IERS-B: {}'
-                                .format(' and '.join(all_urls),
-                                        ';'.join(err_list))))  # noqa
+            warn(AstropyWarning(
+                f'failed to download {" and ".join(all_urls)}, '
+                f'using local IERS-B: {err}'))
```

Comment:
I was told that I no longer need `str(err)` to format it properly? Or do I need a `!r`? Please let me know if you want me to add anything to the `err` formatting.

Reply:
As is, you get the text of whatever the error was, and `str(err)` indeed does exactly the same. With `err!r`, one gets also the type of error. I think since we are warning here, it might only be confusing to also have an error type. In any case, it is most similar to what was there before.

Reply:
Thanks for the clarification!